### PR TITLE
docs: Convert website to use `{pkgdown}`

### PR DESCRIPTION
**Pull request by olivroy**
Proof of concept.

I think some of the information is a bit dated, but will not pursue further if there is no interest.

For deployment, you need to setup gh pages with `usethis::use_pkgdown_github_pages()`.

Everything renders! Polishing probably necessary in a potential follow-up

But having the function reference online, search function seems like an improvement to me?
Capture d’écran, le 2024-02-16 à 9 24 39 a m

Outstanding work if you decide to proceed:

On my side:

* Correct links
* Replace deprecated things
* Are there useful things in `docs/` that is worth saving somewhere else?
* Polishing

On your side:
* Adjust css / look (on you)
* `usethis::use_pkgdown_github_pages()`
* Verify if there is an alternative to "hrbrmstr/albersusa" (archived and uses archived packages)

### CODE REVIEW

File: _pkgdown.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+url: ~
+template:
+  bootstrap: 5
+
```
Comment (schloerke) [inline _pkgdown.yml:4]: If the colors (untested) seem off, please feel free to adjust them. I tried to take them from links and grey background in https://leafletjs.com/ .

```suggestion
url: https://rstudio.github.io/leaflet/

development:
  mode: auto

template:
  package: tidytemplate
  bootstrap: 5
  bslib:
    primary: "#199900"
    navbar-background: "#f8f8f8"
  trailing_slash_redirect: true
```

File: DESCRIPTION [hunk 2]
```diff
@@ -71,12 +71,15 @@ Suggests:
...
     R6,
     RJSONIO,
+    rmarkdown,
     s2,
     sf (>= 0.9-6),
     shiny,
     terra,
     testthat (>= 3.0.0)
 Config/testthat/edition: 3
+Config/Needs/website: geojsonio, ncdf4
```
Comment (schloerke) [inline DESCRIPTION:81]:
```suggestion
Config/Needs/website: geojsonio, ncdf4, tidyverse/tidytemplate
```

File: NEWS.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 * Fixed #893: Correctly call `terra::crs()` when checking the CRS of a `SpatVector` object in `pointData()` or `polygonData()` (thanks @mkoohafkan, #894).
 
+* leaflet now has a brand new pkgdown site (@olivroy, #902)
```
Comment (schloerke) [inline NEWS.md:5]:
```suggestion
* `{leaflet}` now has a brand new pkgdown site (@olivroy, #902)
```

File: inst/examples/highlight-polygons.R [hunk 4]
```diff
@@ -2,8 +2,7 @@
 #' An example to show how to highlight polygons on hover using `onRender`
 #'
 library(sp)
-library(albersusa)
-library(maptools)
+library(albersusa) # Requires rgdal / rgeos archived.
```
Comment (schloerke) [inline inst/examples/highlight-polygons.R:5]: Yes, we should visit these in a follow up PR.

File: vignettes/articles/json.Rmd [hunk 5]
```diff
@@ -1,14 +1,15 @@
 ---
```
Comment (schloerke) [inline vignettes/articles/json.Rmd:1]: Should the `.Rmd` files be placed in `vignettes/FILE.Rmd`? (Not `vignettes/articles/FILE.Rmd`)

After pkgdown make the articles, I believe the current url will be something like `articles/articles/FILE.html`.

If `vignettes/FILE.Rmd` works the same, please move the files accordingly.

Comment (schloerke) [reply vignettes/articles/json.Rmd:1]: ... Wait... If they are under the articles folder, then the vignette will not be present in the package, but will be present in the pkgdown articles?

If so, TIL and this is great to leave as is!

**Review by schloerke — CHANGES_REQUESTED** [review]

**Comment by schloerke** [comment]
Thank you @olivroy !!

Could extra requests:
- [ ] Please uncomment https://github.com/rstudio/leaflet/blob/e15a997421832f6fc98681390f739b99f9237b06/.github/workflows/R-CMD-check.yaml#L16-L18 ; When the PR is merged, I'll redirect the website location.

File: vignettes/articles/basemaps.Rmd [hunk 6]
```diff
@@ -1,7 +1,14 @@
 ---
```
[the remark this answers is not in the document]
Comment (olivroy) [reply vignettes/articles/basemaps.Rmd:1]: The thing is that `vignettes/` go to CRAN by default, while `vignettes/articles/` do not. https://usethis.r-lib.org/reference/use_vignette.html?q=use_artic#ref-usage!

File: vignettes/articles/json.Rmd (hunk 5 above)
Comment (olivroy) [reply vignettes/articles/json.Rmd:1]: Indeed, I only left the intro one as a vignette, because it is pretty light (no image or embedded maps_

**Comment by olivroy** [comment]
@schloerke Would you want me to delete the old docs directory, or leaving it as `docs_old/` is okay for now?

File: _pkgdown.yml (hunk 1 above)
Comment (olivroy) [reply _pkgdown.yml]: You can always readjust css later if you want, but the green makes it look more unique
Capture d’écran, le 2024-02-16 à 10 03 10 a m

File: docs_old/scratch.Rmd
Comment (schloerke) [inline docs_old/scratch.Rmd:1]: Commenting for visibility...

This file is being removed as it was never surfaced in the previous website

**Comment by schloerke** [comment]
> @schloerke Would you want me to delete the old docs directory, or leaving it as `docs_old/` is okay for now?

The geojson files in `docs_old/json` need to be available to the files. Maybe we put the data files in the `pkgdown/assets` folder and add the assets folder in the pkgdown config? (And also move the `_pkgdown.yml` file to `pkgdown/_pkgdown.yml`, similar to https://github.com/rstudio/plumber/blob/56a69494cf15c3c3ca0831a86a91ba033337206a/pkgdown/_pkgdown.yml#L9 )

By having the files as assets, they are available on the website and users can see the full url. In the past, we've had comments that the files do not exist on user's local machines when copying the example to run. So I'd like to keep the URL behavior if possible.

I do not see any other files that I would keep in `docs_old/`. Let's remove it? Thank you!

File: vignettes/articles/basemaps.Rmd (hunk 6 above)
[the remark this answers is not in the document]
Comment (schloerke) [reply vignettes/articles/basemaps.Rmd:1]: Closing as duplicate of other comment.

(Keeping files here!)

File: vignettes/articles/json.Rmd (hunk 5 above)
Comment (schloerke) [reply vignettes/articles/json.Rmd:1]: TIL! Thank you!

File: .Rbuildignore [hunk 7]
```diff
@@ -36,3 +36,5 @@
 ^vignettes/articles$
 ^_pkgdown\.yml$
 ^pkgdown$
+docs_old$
```
Comment (schloerke) [inline .Rbuildignore:39]:
```suggestion
```

File: .Rbuildignore [hunk 8]
```diff
@@ -33,3 +33,8 @@
 ^revdep$
 ^CRAN-SUBMISSION$
 ^_dev$
+^vignettes/articles$
+^_pkgdown\.yml$
```
Comment (schloerke) [inline .Rbuildignore:37]: (Given ./_pkgdown.yml is moved to `./pkgdown/_pkgdown.yml`)

```suggestion
```

File: docs_old/scratch.Rmd
Comment (olivroy) [reply docs_old/scratch.Rmd:1]: Moved it to `scripts`. Although it has not been touched in 6 years, not worth deleting!

**Comment by schloerke** [comment]
Two comments then I think we're good to go!

------------------

Can we move the json files in `pkgdown/assets/FILE` to `pkgdown/assets/json/FILE`? This will keep the existing urls like https://rstudio.github.io/leaflet/json/us-states.geojson working.

------------

We need to add redirect HTML files from the previous vignette locations to the new locations. In addition, I'd like to maintain the hash being used. (**working on getting an example file / instructions)

**Comment by schloerke** [comment]
We should add files with the content of: (untested)

```html
<!DOCTYPE html>
<html>
  <body>
    <h1>VIGTITLE</h1>
    Please find the article at its <a href="articles/ARTICLE">new location</a>
<script>
  window.location.href = "articles/ARTICLE" + window.location.hash;
</script>
  </body>
</html>
```

Where `VIGTITLE` is the article title and `ARTICLE` is the file name.

Within `pkgdown/assets`, we need an HTML redirect file (named `ARTICLE`) for every past article (except for `./`):

`ARTICLE` - `VIGTITLE`:
* `map_widget.html` - `The Map Widget`
* `basemaps.html` - `Basemaps`
* `markers.html` - `Markers`
* `popups.html` - `Popups and Labels`
* `shapes.html` - `Lines and Shapes`
* `json.html` - `GeoJSON and TopoJSON`
* `raster.html` - `Raster Images`
* `shiny.html` - `Shiny Integration`
* `colors.html` - `Colors`
* `legends.html` - `Legends`
* `showhide.html` - `Show/Hide Layers`
* `choropleths.html` - `Choropleths`
* `projections.html` - `Projections`
* `morefeatures.html` - `Additional Features`

File: docs_old/scratch.Rmd
Comment (schloerke) [reply docs_old/scratch.Rmd:1]: Thank you!

**Comment by olivroy** [comment]
I tested the pkgdown approach locally, but ideally, we'd overwrite the site now, and move to dev mode later?
I tested this with servr locally, and changed the URL manually in my browser, and it works well!

File: DESCRIPTION [hunk 9]
```diff
@@ -71,12 +71,15 @@ Suggests:
...
     R6,
     RJSONIO,
+    rmarkdown,
     s2,
     sf (>= 0.9-6),
     shiny,
     terra,
     testthat (>= 3.0.0)
 Config/testthat/edition: 3
+Config/Needs/website: geojsonio, ncdf4, tidyverse/tidytemplate
```
Comment (schloerke) [inline DESCRIPTION:81]:
```suggestion
Config/Needs/website: dplyr, geojsonio, ncdf4, tidyverse/tidytemplate
```

**Review by schloerke — APPROVED** [review]

**Comment by schloerke** [comment]
Thank you @olivroy !

**Comment by olivroy** [comment]
Thanks for the prompt review!  (I would have forgotten the redirects !)

I will send a follow-up early next week to address outstanding issues! And will test outstanding issues, reorder vignettes etc. maybe use the json approach and move other shapefiles / resources to `pkgdown/assets` to read from the website as much as possible, to make the reproducibility of examples easier to do for users.

**Comment by schloerke** [comment]
> ( I didn't want to read the content in the context of this PR)

I appreciate it. 

I look forward to the followup next week. I'll try to have the PR merged (**waiting on team approval) later today.

**Comment by schloerke** [comment]
Thank you @olivroy !!

**Pull request merged**